skimage/segmentation/_clear_border.py

```diff
@@ -71,14 +65,8 @@ def clear_border(labels, buffer_size=0, bgval=0, in_place=False, mask=None,
         # ignore buffer_size if mask
         raise ValueError("buffer size may not be greater than labels size")
 
-    if out is not None:
-        np.copyto(out, labels, casting='no')
-        in_place = True
-
-    if not in_place:
+    if out is None:
         out = labels.copy()
-    elif out is None:
-        out = labels
```

Comment:
Please have a look if I reduced this logic correctly. I find it quite confusing and am not sure whether I've missed something or there was a bug before...